### PR TITLE
Ws 5.5 user profile page

### PR DESCRIPTION
**Pull request by natashabuckham**
Backend:
change the /users/{username} endpoint to be /users/profile
use getbyid in the user service instead of getbyusername
pass the id from the Claims Principal object from the controller to the service
make a Profile Response model to return all user details except id and password
‌
Frontend:
make a new Profile component in the Pages folder
Display user details
Add Update button 
Add Delete button
Make a fetch request in backendClient.ts to link the /users/profile endpoint - pass JWT from loginContext
Add ‘Profile’ link to the navigation links
Add conditional logic to only show this Profile link if the user is logged in - get logged in status from loginContext
Add styling
Add 'unauthorized' message if not logged in and user tries to access the url manually

### CODE REVIEW

File: backend/Models/Response/ProfileResponse.cs
Comment (sun-amy) [inline backend/Models/Response/ProfileResponse.cs:1]: I don't think the UserName and email should be nullable here.

**Review by sun-amy — COMMENTED** [review]
Makes sense

File: frontend/src/api/backendClient.ts [hunk 1]
```diff
@@ -1,3 +1,12 @@
+export interface User {
+  userName: string
+  firstName: string
+  lastName: string
+  email: string
+  totalPointsEarned: number
+  aboutMe: string
```
Comment (Kiran-R-K) [inline frontend/src/api/backendClient.ts:7]: It looks like none of these fields are nullable here but in the `ProfileResponse` that's being mapped onto this object some of the fields are nullable - I think and they should match

File: backend/Models/Response/ProfileResponse.cs
Comment (Kiran-R-K) [reply backend/Models/Response/ProfileResponse.cs:1]: Both those fields are nullable on the database - in an ideal world we would fix that to make them not nullable. However when a user registers they HAVE to provide a username and email so barring edge cases we can assume that every user will have a username and password and we should have those fields be not nullable in the models.
If we had more time we would fix this in the database too

**Review by Kiran-R-K — CHANGES_REQUESTED** [review]

File: frontend/src/api/backendClient.ts (hunk 1 above)
Comment (natashabuckham) [reply frontend/src/api/backendClient.ts:7]: All fixed :)

**Review by Kiran-R-K — APPROVED** [review]

**Pull request merged**